### PR TITLE
Small CSR tweaks

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
Addressing some review comments from https://github.com/kubermatic/kubeone/pull/1750#pullrequestreview-864477584

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by ahmedwaleedmalik — APPROVED** [review]
/lgtm 
/approve

File: pkg/tasks/certs.go [hunk 1]
```diff
@@ -235,26 +234,26 @@ func approvePendingCSR(s *state.State, node *kubeoneapi.HostConfig, conn ssh.Con
 
 func validateCSR(spec certificatesv1.CertificateSigningRequestSpec, node *kubeoneapi.HostConfig) error {
 	if fmt.Sprintf("%s:%s", nodeUser, node.Hostname) != spec.Username {
-		return errors.New("")
+		return fmt.Errorf("CSR username %q and node hostname %q do not match", spec.Username, node.Hostname)
```
Comment (xmudrii) [inline pkg/tasks/certs.go:237]: I guess this should be `errors.New` instead.

Comment (kron4eg) [reply pkg/tasks/certs.go:237]: I like Errorf better since it manifests that "format" is used

Comment (xmudrii) [reply pkg/tasks/certs.go:237]: There’s also Errorf in the errors package, so we get to use same package everywhere. Just a nit though.

**Review by xmudrii — APPROVED** [review]

**Pull request merged**